### PR TITLE
RHCLOUD-23167 Adding tests for EndpointReadyChecker

### PR DESCRIPTION
**Pull request by josejulio**
Adding some tests for the `EndpointReadyChecker`. 
The tests covers the add (success/failed) and deleting scenarios.

### CODE REVIEW

File: backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointReadyCheckerTest.java [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+        final String PROCESSOR_4 = "processor-multiple-4";
+        final String PROCESSOR_5 = "processor-multiple-5";
+
+        mockAddProcessor(PROCESSOR_1);
+        Map<String, Object> jsonEndpoint1 = createEndpoint(identityHeader);
+
+        mockAddProcessor(PROCESSOR_2);
+        Map<String, Object> jsonEndpoint2 = createEndpoint(identityHeader);
+
+        // We are going to delete the first 2 processors - fishing their lifecycle to have them as ready.
```
Comment (josejulio) [inline backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointReadyCheckerTest.java:186]:
```suggestion
        // We are going to delete the first 2 processors - finishing their lifecycle to have them as ready.
```

File: backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointReadyCheckerTest.java [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+        endpointReadyChecker.execute();
+
+        // delete it now
+        deleteEndpoint(identityHeader, jsonEndpoint.get("id").toString());
+
+        // Verify it's marked for deletion
+        Endpoint endpoint = resourceHelpers.getEndpoint(ORG_ACCOUNT_ID, UUID.fromString(jsonEndpoint.get("id").toString()));
+        assertEquals(EndpointStatus.DELETING, endpoint.getStatus());
+
+        processor.setStatus("deleted");
```
Comment (josejulio) [inline backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointReadyCheckerTest.java:157]:
```suggestion
        // Update the response to deleted
        processor.setStatus("deleted");
```

**Comment by josejulio** [comment]
Updated to consider http error code throw when calling the `get-processor` api

**Review by gwenneg — APPROVED** [review]
LGTM

**Pull request merged**